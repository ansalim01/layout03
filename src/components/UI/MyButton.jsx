import React from "react";
import styles from "./MyButton.module.scss";

function MyButton(props) {
  let st = styles.buttonIm;

  if (props.st === "buttonTx") {
    st = styles.buttonTx;
  }
  return (
    <div className={styles.buttons} onClick={props.onClick}>
      <button className={st}>
        <span>{props.text}</span>
        {props.children}
      </button>
    </div>
  );
}

export default MyButton;
