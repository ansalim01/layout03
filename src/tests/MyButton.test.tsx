import React from "react";
import { render, screen } from "@testing-library/react";
import MyButton from "../components/UI/MyButton";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";

describe("MyButton", () => {
  test("MyButton", () => {
    render(<MyButton />);
    const linkElement = screen.getByTestId("custom-button");
    expect(linkElement).toMatchSnapshot();
  });
});
