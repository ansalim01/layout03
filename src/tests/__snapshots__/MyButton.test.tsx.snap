// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton MyButton 1`] = `
<div
  class="buttons"
  data-testid="custom-button"
>
  <button
    class="buttonIm"
  >
    <span />
  </button>
</div>
`;

exports[`test app renders learn react link 1`] = `
<div
  class="buttons"
  data-testid="custom-button"
>
  <button
    class="buttonIm"
  >
    <span />
  </button>
</div>
`;
